internal/log/log.go

```diff
@@ -0,0 +1,22 @@
+// Package log is a preflight-local logrus bridge, removing the
+// dependency on logrus' global logger instance.
+package log
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+var l *logrus.Logger
+
+// Logger returns the configured logger, or a new unconfigured logger
+// if one has not already been configured.
+func Logger() *logrus.Logger {
+	if l == nil {
+		l = logrus.New()
+	}
+
+	return l
+}
+
+// L is a convenience alias to Logger.
+var L = Logger
```

Comment:
nit: this `log.L().Stuff()` looks ugly ... may be not worth implementing an interface so that we can consume logging like `log.Stuff()` but looks less ugly imho like in [opcap](https://github.com/opdev/opcap/blob/main/internal/logger/logger.go)

Reply:
actually would say not a nit 🧐

Reply:
I considered this - but It feels a bit unnecessary, and like quite a bit of overhead, to add all of `logrus.Logger`'s methods as package-level functions just to achieve the same API we were getting with the import of logrus itself.

I suppose I can make it so if we deem it truly beneficial, but I do feel this achieves the same result without having N (where N == the number of methods `logrus.Logger` has) additional functions in our own code. We would need something like Warn/Info/Error/Debug/Trace + their formatted string variants (e.g. `Infof`), SetOutput, SetLevel, SetFormatter... and each of these would then technically need their own test to not impact coverage.

In theory, I could have just allowed access to the `l` variable itself, but then I'd have to mess with init functions. Ultimately, this felt cleaner (of the options).

Let me know.

Reply:
Going the opcap route seems overkill, and also confusing for our use case. Why add overhead and methods that we have to maintain?

Reply:
Does not have to be everything just what we use. I was thinking from the angle of the rest of code being decoupled from Logrus so that we can swap in a different logger in the future with minimal impact. However, I imagine most loggers have a similar interface so may not be worth it. On the other hand having a way to get a reference to the logger like `logger = log.GetLogger()` would let consumers do logger.Stuff() too right ?

Reply:
Because this package is internal, we would be the only consumers.

Reply:
By consumers I don't mean users, just things that use the log package. It could be "decoupled" if we only referenced Logrus specific things in the log package and everything else (all consumers) only consume what the log package exposes. Yeah its internal but we still have a "global" logger i.e. everything that consumes logging from the log package uses the same instance regardless of `log.L().Stuff();` vs `logger = log.Logger(); logger.stuff();` because of the "singleton" initialization.

Reply:
> Does not have to be everything just what we use.
We have a flag exposed to the caller for them to use how they would like, which means we'd have to code everything logrus supports, which is overkill IMO.

Reply:
The definition of a logging interface that we then consume definitely has merit - and very well may be a part of the logging refactor I described in the original message. That said, this is not intended to be that refactor, but is rather just a bugfix.

Reply:
Fair points but what is the case for `log.L().Stuff()`; vs `logger = log.Logger(); logger.stuff();` ?

Reply:
The former option `log.L().Stuff()` saves a variable assignment. Otherwise, those are identical.

Reply:
One is more ugly 😝. But can safely downgrade this conversation back to nit. I didn't read the PR description fully 😅.

Reply:
I think the core of what you were asking is 100% on the money! Make our internal/log package the _only_ import of our code. We did similar for viper a while back.

I just don't know if I want to do it _here_. We're on a branch here that hasn't merged to main yet. I'm trying to make changes only scoped to the end-goal of the library becoming published. The logging refactor is imminent, don't you worry.

Reply:
I still think the library should use https://github.com/go-logr/logr and leave the actual log implementation as a detail for the user. As long as the logger given supports that interface, win.

Reply:
> For what it's worth - we're not actually decoupled from logrus here - we're still very dependent on logrus. This just lets us avoid clobbering another consumer's global logrus.Logger configuration because we _internally_ refer to our own.

I'm good with this being a stop-gap, knowing that we still want a major log refactor.